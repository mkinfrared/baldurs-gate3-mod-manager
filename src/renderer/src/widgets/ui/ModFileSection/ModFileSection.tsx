import { ModFileForm } from "@renderer/features/modFile";
import { classNames } from "@renderer/shared/lib/helpers";
import { Heading, StickyBlock } from "@renderer/shared/ui";

import css from "./ModFileSection.module.scss";
import { ModFileSectionProps } from "./ModFileSection.type";

const ModFileSection = ({ className, game }: ModFileSectionProps) => (
  <div
    className={classNames(css.ModFileSection, className)}
    data-testid="ModFileSection"
  >
    <StickyBlock>
      <Heading>Modification Files</Heading>
    </StickyBlock>
    <ModFileForm game={game} />
  </div>
);

export { ModFileSection };
