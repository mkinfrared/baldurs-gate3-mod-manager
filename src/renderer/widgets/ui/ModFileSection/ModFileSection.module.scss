.ModFileSection {
  display: flex;
  flex-direction: column;
}
