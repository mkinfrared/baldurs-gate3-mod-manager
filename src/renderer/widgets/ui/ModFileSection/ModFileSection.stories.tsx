import { Meta, StoryObj } from "@storybook/react";

import { ModFileSection } from "./ModFileSection";
import css from "./Story.module.scss";

type Story = StoryObj<typeof meta>;

const meta = {
  title: "WIDGETS/ModFileSection",
  component: ModFileSection,
  tags: ["autodocs"],
} satisfies Meta<typeof ModFileSection>;

const Default: Story = {
  args: {},
  render: (args) => (
    <div className={css.Story}>
      <ModFileSection {...args} />
    </div>
  ),
};

export { Default };

export default meta;
