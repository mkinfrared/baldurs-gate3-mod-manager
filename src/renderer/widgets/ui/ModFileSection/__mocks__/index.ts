export { ModFileSectionMock as ModFileSection } from "./ModFileSection.mock";
