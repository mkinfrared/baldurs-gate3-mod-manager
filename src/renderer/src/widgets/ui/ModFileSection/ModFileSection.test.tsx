import { render } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";

import { ModFileSection } from "./ModFileSection";

vi.mock("@renderer/features/modFile");

describe("<ModFileSection />", () => {
  const game = "BG3";
  const Component = <ModFileSection game={game} />;

  it("should be defined", () => {
    expect(ModFileSection).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("ModFileSection");

    expect(element).toBeDefined();
  });
});
