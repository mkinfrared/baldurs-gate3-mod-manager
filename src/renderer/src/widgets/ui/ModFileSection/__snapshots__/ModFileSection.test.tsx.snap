// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ModFileSection /> > should match the snapshot 1`] = `
<div>
  <div
    class="ModFileSection"
    data-testid="ModFileSection"
  >
    <div
      class="StickyBlock"
      data-testid="StickyBlock"
    >
      <h3
        class="Heading"
        data-testid="Heading"
      >
        Modification Files
      </h3>
    </div>
  </div>
</div>
`;
